docs/11.EXT-REFERENCE-MODULE.md

```diff
@@ -0,0 +1,116 @@
+#  Modules API
+
+## jerryx_module_resolve
+
+**Summary**
+
+Load a copy of a module into the current context or return one that was already loaded if it is found.
+
+**Prototype**
+
+```c
+jerry_value_t
+jerryx_module_resolve (const jerry_char_t *name);
+```
+
+- `name` - the name of the module to load.
+- return value - `jerry_value_t` representing the module that was loaded, or the error that occurred in the process.
+
+
+# Modules data types
+
+## jerryx_module_t
+
+**Summary**
+
+Structure to define a module
+
+**Prototype**
+
+```c
+typedef struct
+{
+  jerry_char_t *name; /**< name of the module */
+  jerry_value_t (*on_resolve)(void); /**< function that returns a new instance of the module */
+} jerryx_module_t;
+```
+
+# Modules helper macros
+
+## JERRYX_MODULE
+
+**Summary**
+
+Helper macro to define a module. Declares a variable of type `jerryx_module_t` as global static and places it into the
+appropriate linker section of the executable.
+
+**Note**: The helper macro must appear at the bottom of a source file, and no semicolon must follow it.
+
+**Prototype**
+```c
+#define JERRYX_MODULE(module_name, on_resolve_cb)
+```
+
+- `module_name` - the name of the module without quotes
+- `on_resolve_cb` - the function that will be called when the module needs to be loaded.
+
+**Example**
+
+```c
+#include "jerryscript.h"
+#include "jerryscript-ext/module.h"
+
+static jerry_value_t
+my_module_on_resolve (void)
+{
+  return jerry_create_external_function (very_useful_function);
+} /* my_module_on_resolve */
+
+/* Note that there is no semicolon at the end of the next line. This is how it must be. */
+JERRYX_MODULE (my_module, my_module_on_resolve)
+```
+
+**See also**
+
+- [jerryx_module_t](#jerryx_module_t)
+
+
+## JERRYX_MODULE_RESOLVER
+
+**Summary**
+
+Define a module resolver. An invocation of this macro is to be followed by the body of a module resolver. The macro
+declares a function of the name given in `cb_name` which accepts a single parameter `const jerry_char_t *name`. It is
+expected that the function returns a non-zero value of type `jerry_value_t` if the resolver is able to associate `name`
+with such a value, or zero if it is not. If the function returns non-zero `jerryx_module_resolve ()` will cache the
```

Comment:
Nevertheless, I can change the signature to return `bool` and the result in an out parameter.